protocol-units/da/movement/protocol/util/src/blob/ir/data.rs

```diff
@@ -27,11 +27,11 @@ where
 		Self { blob, timestamp, __curve_marker: std::marker::PhantomData }
 	}
 
-	pub fn now(blob: Vec<u8>) -> Result<Self, anyhow::Error> {
-		Ok(Self::new(
-			blob,
-			std::time::SystemTime::now().duration_since(std::time::UNIX_EPOCH)?.as_secs(),
-		))
+	pub fn now(blob: Vec<u8>) -> Self {
+		// mark the timestamp as now in milliseconds
+		let timestamp = chrono::Utc::now().timestamp_micros() as u64;
```

Comment:
Perhaps it's good to document that the `timestamp` field of `InnerSignedBlobV1Data` is the Unix tiemstamp in microseconds, to forestall this kind of coding mistakes from arising.

Even better would be to actually use a dedicated timestamp type with controlled conversions, but let's do baby steps.

Reply:
Done